### PR TITLE
Upgrade template

### PR DESCRIPTION
**Pull request by pacrob**
### What was wrong?

Needs merging with python project template.

### How was it fixed?

Merged with template.
 - Dropped python 3.7 support
 - Use `pre-commit` for linting

### Todo:
- [x] Update the CHANGELOG with appropriate descriptions reflecting the changes in this PR

#### Cute Animal Picture

image

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -24,27 +25,40 @@ clean-pyc:
...
+		echo "\n\n\n * pre-commit should have fixed the errors above. Running again to make sure everything is good..." \
+		&& pre-commit run --all-files --show-diff-on-failure \
+	)
 
 test:
 	pytest tests
 
-test-all:
-	tox
+build-docs: docs
```
Comment (fselmo) [inline Makefile]: This doesn't seem useful if it just calls `docs` 🤔... does this facilitate something I'm not seeing when the template is applied?

Comment (pacrob) [reply Makefile]: It's there so that the `notes` and `release` commands don't need to change from the template.

Comment (pacrob) [reply Makefile]: Changed it, won't be an issue in future template merges, and looks cleaner.

Comment (fselmo) [reply Makefile]: I figured it was something like that. It seems fine tbh, as long as we do it similarly on all CHANGELOG repos so we don't start deviating too much from a known pattern. I believe `py-geth` is set up this way. Can we mimic that repo here?

Comment (fselmo) [reply Makefile]: > Changed it, won't be an issue in future template merges, and looks cleaner.

Oops I missed this when I replied. That was quick. That makes sense. We should change it in *py-geth* as well when we get the chance.

File: pyproject.toml [hunk 2]
```diff
@@ -1,2 +1,122 @@
...
+# summary line.  Neither one of those requirements seem appropriate.
+
+[tool.pytest.ini_options]
+addopts = "-v --showlocals --durations 10"
+xfail_strict = "True"
+log_format = "%(levelname)8s  %(asctime)s  %(filename)20s  %(message)s"
+log_date_format = "%m-%d %H:%M:%S"
+
+[tool.towncrier]
+# Read https://github.com/ethereum/eth-bloom/blob/main/newsfragments/README.md for instructions
```
Comment (fselmo) [inline pyproject.toml]:
```suggestion
# Read https://github.com/ethereum/eth-bloom/blob/master/newsfragments/README.md for instructions
```

see: [eth-typing example](https://github.com/ethereum/eth-typing/blob/master/newsfragments/README.md)

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+### How was it fixed?
+
+### Todo:
+
+- [ ] Clean up commit history
+
+- [ ] Add or update documentation related to these changes
+
+- [ ] Add entry to the [release notes](https://github.com/ethereum/eth-bloom/blob/main/newsfragments/README.md)
```
Comment (fselmo) [inline .github/pull_request_template.md]:
```suggestion
- [ ] Add entry to the [release notes](https://github.com/ethereum/eth-bloom/blob/master/newsfragments/README.md)
```

for now

File: tox.ini [hunk 4]
```diff
@@ -1,43 +1,65 @@
...
 usedevelop=True
 commands=
-    pytest tests
-extras=
-    test
-deps =
-    eth-hash[pycryptodome]
-basepython =
-    py37: python3.7
+    core: pytest {posargs:tests/core}
```
Comment (fselmo) [inline tox.ini]: Same issue we saw recently here. There are tests that exist outside of `tests/core` within `tests`. This is only testing that the library can be imported at the moment. Let's change this to the below perhaps:

```suggestion
    core: pytest {posargs:tests}
```

I'm also seeing a warning that might be nice to silence in the core tests related to `python_paths` within `pytest.ini`. Let's remove that line from that file.

File: pyproject.toml (hunk 2 above)
Comment (pacrob) [reply pyproject.toml]: curse you main!

Comment (pacrob) [reply pyproject.toml]: fixed

File: .github/pull_request_template.md (hunk 3 above)
Comment (pacrob) [reply .github/pull_request_template.md]: fixed

File: tox.ini (hunk 4 above)
Comment (pacrob) [reply tox.ini]: Another oversight - `pytest.ini` should be deleted, as all the config has moved to `pyproject.toml`. It is now deleted.

**Review by fselmo — APPROVED** [review]

**Pull request merged**